internal/trigger/reader/reader.go

```diff
@@ -68,107 +65,44 @@ type reader struct {
 	elReader map[vanus.ID]struct{}
 	events   chan<- info.EventRecord
 	stop     context.CancelFunc
-	stctx    context.Context
 	wg       sync.WaitGroup
-	lock     sync.Mutex
 }
 
 func NewReader(config Config, events chan<- info.EventRecord) Reader {
-	if config.CheckEventLogInterval <= 0 {
-		config.CheckEventLogInterval = checkEventLogInterval
-	}
 	config.SubscriptionIDStr = config.SubscriptionID.String()
 	r := &reader{
 		config:   config,
 		events:   events,
 		elReader: make(map[vanus.ID]struct{}),
+		stop:     func() {},
```

Comment:
the `stop` will be called in the function close, it will panic if the function start doesn't call

Reply:
it it better ?`if r.stop != nil {r.stop()}`

Reply:
yes, it's no problem, but I need to add an if check, I think it's a code style. And `stop` can't be nil because in the function start it is initialized at the first line. I will remove it too.